sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/AcsRouterJobClassificationFailedEventData.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Azure.Core;
+
+namespace Azure.Messaging.EventGrid.SystemEvents
+{
+    /// <summary> Schema of the Data property of an EventGridEvent for a Microsoft.Communication.RouterJobClassificationFailed event. </summary>
+    public partial class AcsRouterJobClassificationFailedEventData
+    {
+        /// <summary> List of Router Communication Errors. </summary>
+        [CodeGenMember("Errors")]
+        internal IReadOnlyList<AcsRouterCommunicationError> ErrorsInternal { get; }
+
+        /// <summary> List of Router Communication Errors. </summary>
+        public IReadOnlyList<ResponseError> Errors
+        {
+            get
+            {
+                if (_errors == null)
+                {
+                    // Need to re-serialize to be able to deserialize as ResponseError with the internal properties populated.
```

Comment:
Sorry if this is a gap in my understanding of ResponseError/generation - but it looks like `AcsRouterCommunicationError` and `ResponseError` are the same, why not just use `ResponseError` everywhere instead?

Reply:
The AcsRouterCommunicationError is generated by code gen. We don't want to expose this type and want to instead use ResponseError. For data plane, this is the only way to do it. Also, ResponseError doesn't have a public ctor that allows you to pass the InnerError, Target, etc, so we need to get the serialized payload and rely on the deserialization converter of ResponseError.

Reply:
Got it- thank you!